### PR TITLE
feat: add repository visibility (public/private) to output

### PR DESCRIPTION
**Pull request by jmeridth**
# Pull Request

## Proposed Changes

Our priority was to archive public repositories that were not updated in a certain amount of inactive days.  This PR adds that information to the output.

- [x] update code/tests for new visibility column

### Author/Contributor

- [x] If documentation is needed for this change, has that been included in this pull request
- [x] run `make lint` and fix any issues that you have introduced
- [x] run `make test` and ensure you have test coverage for the lines you are introducing

### Reviewer

- [x] Label as either `bug`, `documentation`, `enhancement`, `infrastructure`, or `breaking`

### CODE REVIEW

File: stale_repos.py [hunk 1]
```diff
@@ -191,10 +195,12 @@ def write_to_markdown(inactive_repos, inactive_days_threshold, file=None):
...
         )
-        file.write("| Repository URL | Days Inactive | Last Push Date |\n")
-        file.write("| --- | --- | ---: |\n")
-        for repo_url, days_inactive, last_push_date in inactive_repos:
-            file.write(f"| {repo_url} | {days_inactive} | {last_push_date} |\n")
+        file.write("| Repository URL | Days Inactive | Last Push Date | Visibility |\n")
+        file.write("| --- | --- | --- | ---: |\n")
+        for repo_url, days_inactive, last_push_date, private in inactive_repos:
+            file.write(
+                f"| {repo_url} | {days_inactive} | {last_push_date} | {private} |\n"
```
Comment (zkoppert) [inline stale_repos.py:202]:
```suggestion
                f"| {repo_url} | {days_inactive} | {last_push_date} | {visibility} |\n"
```

File: stale_repos.py [hunk 2]
```diff
@@ -218,12 +225,13 @@ def output_to_json(inactive_repos, file=None):
...
     # ]
     inactive_repos_json = []
-    for repo_url, days_inactive, last_push_date in inactive_repos:
+    for repo_url, days_inactive, last_push_date, private in inactive_repos:
         inactive_repos_json.append(
             {
                 "url": repo_url,
                 "daysInactive": days_inactive,
                 "lastPushDate": last_push_date,
+                "visibility": private,
```
Comment (zkoppert) [inline stale_repos.py:234]:
```suggestion
                "visibility": visibility,
```

Comment (jmeridth) [reply stale_repos.py]: Added fix and set you as co-author.  Thank you for feedback

File: stale_repos.py [hunk 3]
```diff
@@ -218,12 +225,13 @@ def output_to_json(inactive_repos, file=None):
     #   }
     # ]
     inactive_repos_json = []
-    for repo_url, days_inactive, last_push_date in inactive_repos:
+    for repo_url, days_inactive, last_push_date, private in inactive_repos:
```
[the remark this answers is not in the document]
Comment (jmeridth) [reply stale_repos.py]: Added fix and set you as co-author.  Thank you for feedback

File: stale_repos.py (hunk 1 above)
Comment (jmeridth) [reply stale_repos.py]: Added fix and set you as co-author.  Thank you for feedback

File: stale_repos.py [hunk 4]
```diff
@@ -132,8 +133,11 @@ def get_inactive_repos(github_connection, inactive_days_threshold, organization)
 
         active_date_disp = active_date.date().isoformat()
         days_inactive = (datetime.now(timezone.utc) - active_date).days
+        private = "private" if repo.private else "public"
         if days_inactive > int(inactive_days_threshold) and not repo.archived:
-            inactive_repos.append((repo.html_url, days_inactive, active_date_disp))
+            inactive_repos.append(
+                (repo.html_url, days_inactive, active_date_disp, private)
```
[the remark this answers is not in the document]
Comment (jmeridth) [reply stale_repos.py]: Added fix and set you as co-author.  Thank you for feedback

**Comment by jmeridth** [comment]
ok.  really broke it.  fixing locally and getting my linting check working locally to match remote. 😄

**Comment by jmeridth** [comment]
Removed the line that I forgot to remove.  Now the linter (in GitHub Actions) for [this line](https://github.com/github/stale-repos/actions/runs/8102140106/job/22143964259#step:4:911) is saying the closing quote needs to be on a new line.  Local syntax does not like that.  I can't run `make test` or `make lint` without it being on line 910 to close the quoted string. 🤔 🤞 this latest run will pass. 😄

**Comment by zkoppert** [comment]
Teamwork! Wohoo! 🎉

**Comment by jmeridth** [comment]
This will be the 2nd time I've forgotten that GitHub Actions doesn't show `\n`, it does an actual newline. 🤦  Thank you for getting this closed out @zkoppert.  🙇

**Pull request merged**